hermes-common/src/main/java/pl/allegro/tech/hermes/common/broker/KafkaOffsetsStorage.java

```diff
@@ -0,0 +1,93 @@
+package pl.allegro.tech.hermes.common.broker;
+
+import jersey.repackaged.com.google.common.collect.Lists;
+import kafka.common.OffsetAndMetadata;
+import kafka.common.OffsetMetadataAndError;
+import kafka.common.TopicAndPartition;
+import kafka.javaapi.OffsetCommitRequest;
+import kafka.javaapi.OffsetCommitResponse;
+import kafka.javaapi.OffsetFetchRequest;
+import kafka.javaapi.OffsetFetchResponse;
+import kafka.network.BlockingChannel;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.api.TopicName;
+import pl.allegro.tech.hermes.common.time.Clock;
+
+import java.util.LinkedHashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Consumer;
+
+public class KafkaOffsetsStorage implements OffsetsStorage {
+
+    private static final Logger logger = LoggerFactory.getLogger(KafkaOffsetsStorage.class);
+    private static final String EMPTY_METADATA = "";
+    private static final String CLIENT_ID = "client";
+    private static final int CORRELATION_ID = 0;
+    private static final short VERSION = 1;
+
+    private final BlockingChannelFactory blockingChannelFactory;
+    private final Clock clock;
+
+    public KafkaOffsetsStorage(BlockingChannelFactory blockingChannelFactory, Clock clock) {
+        this.blockingChannelFactory = blockingChannelFactory;
+        this.clock = clock;
+    }
+
+    @Override
+    public void setSubscriptionOffset(TopicName topicName, String subscriptionName, int partitionId, long offset) {
+        String groupId = Subscription.getId(topicName, subscriptionName);
+        BlockingChannel channel = blockingChannelFactory.create(groupId);
```

Comment:
BlockingChannel is created here for every offset commit (and offset read). I think it's inefficient because BrokerChannel makes connection to broker so better will be if we use some cache to create only once connection instead of every time for commit offset.